layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -1350,10 +1350,43 @@ the right."
   (interactive)
   (set-buffer-file-coding-system 'undecided-dos nil))
 
-(defun spacemacs/copy-file ()
-  "Write the file under new name."
-  (interactive)
-  (call-interactively 'write-file))
+(defun spacemacs/copy-file (arg)
+  "Write the file under new name and select action on new file.
+This command prompts for a new filename then prompts for
+selecting an action to perform on the new file.
+
+If prefixed with the universal ARG \\[universal-argument] the
+prompt includes the buffer file name.
+
+When prompted for selecting an action, pressing `?' will open the
+help menu that explains the different options."
+  (interactive "P")
+  (let* ((new-file-name (read-file-name "Save file as: "
+                                        (if arg buffer-file-name default-directory))))
+    ;; when selecting a directory name, add (buffer-name) or message error for
+    ;; non-visiting file buffers
+    (when (directory-name-p new-file-name)
+      (if buffer-file-name
+          (setq new-file-name (concat new-file-name (buffer-name)))
+        (error "Copying of non-visiting file buffers requires to specify new file name.")))
```

Comment:
"non-visiting file" is not easy to understand

Reply:
I agree somewhat, although I think it is hard to misunderstand this (of course it is "non-visiting file buffer").
I would be happy if you could suggest a better term (English is not my native language).

Reply:
Maybe "non-visiting-file buffers"?

Reply:
this feels like romance language, not english